### DISCUSSION TITLE
Редактирование конфига через pre-build.sh

### DISCUSSION THREAD

**Discussion by kilroy98** [shvchk/padavan-builder-workflow#25] [Вопросы и ответы] [answered]
Появился вопрос, возможно ли через pre-build.sh отредактировать kernel-3.4.x.config? Если да, то как это сделать?
Для примера: заменить `CONFIG_RALINK_RAM_SIZE=128` на `CONFIG_RALINK_RAM_SIZE=512`.

**Comment by shvchk** [comment, marked as answer]
Конечно. Например, используя `sed` (на примере Xiaomi Mi-Mini):

```sh
sed -i 's/^CONFIG_RALINK_RAM_SIZE.*/CONFIG_RALINK_RAM_SIZE=512/' padavan-ng/trunk/configs/boards/XIAOMI/MI-MINI/kernel-3.4.x.config
```

**Reply by kilroy98** [reply]
Спасибо за ответ. Я ведь правильно сделал или для второй и третьей строки нужно иначе делать?
```
sed -i 's/^CONFIG_RALINK_RAM_SIZE.*/CONFIG_RALINK_RAM_SIZE=512/' padavan-ng/trunk/configs/boards/XIAOMI/RM-AC2100/kernel-3.4.x.config
sed -i 's/^CONFIG_RT2880_DRAM_128M=y*/# CONFIG_RT2880_DRAM_128M is not set/' padavan-ng/trunk/configs/boards/XIAOMI/RM-AC2100/kernel-3.4.x.config
sed -i 's/^# CONFIG_RT2880_DRAM_512M is not set*/CONFIG_RT2880_DRAM_512M=y/' padavan-ng/trunk/configs/boards/XIAOMI/RM-AC2100/kernel-3.4.x.config
```

**Reply by shvchk** [reply]
Мне кажется, 2 и 3 строки в сумме эквивалентны одной простой замене:

```sh
sed -i 's/^CONFIG_RT2880_DRAM_128M=y/CONFIG_RT2880_DRAM_512M=y/' padavan-ng/trunk/configs/boards/XIAOMI/RM-AC2100/kernel-3.4.x.config
```

**Reply by kilroy98** [reply]
Спасибо. Просто я вот думаю не сломается ли ничего при сборке, если вот так просто заменить, но при этом оставить `# CONFIG_RT2880_DRAM_512M is not set` и не добавлять `# CONFIG_RT2880_DRAM_128M is not set`

**Reply by shvchk** [reply]
Думаю, закомментированные строки (а `#` в начале строки обозначает именно это) игнорируются при сборке.

**Reply by kilroy98** [reply]
Понял, спасибо.
